### PR TITLE
mempool: adjust the rule of conflicting transaction ranking

### PR DESCRIPTION
**Pull request by AnnaShaleva**
Pay for all the conflicts if you'd like to went in. Close https://github.com/nspcc-dev/neo-go/issues/3028.

### CODE REVIEW

File: pkg/core/mempool/mem_pool_test.go [hunk 1]
```diff
@@ -448,6 +454,12 @@ func TestMempoolAddRemoveConflicts(t *testing.T) {
 		require.Equal(t, false, ok)
 		return tx
 	}
+	getConflictsTx := func(netFee int64, hashes ...util.Uint256) *transaction.Transaction {
+		return getTx(netFee, sender, hashes...)
+	}
+	getMalisiousTx := func(netFee int64, hashes ...util.Uint256) *transaction.Transaction {
```
Comment (roman-khimov) [inline pkg/core/mempool/mem_pool_test.go:460]: malicious

File: pkg/core/mempool/mem_pool_test.go [hunk 2]
```diff
@@ -528,26 +540,88 @@ func TestMempoolAddRemoveConflicts(t *testing.T) {
...
+	// tx15 conflicts with tx14, but added firstly and has the same network fee => tx14 must not be added.
 	tx14 := getConflictsTx(smallNetFee)
 	tx15 := getConflictsTx(smallNetFee, tx14.Hash())
 	require.NoError(t, mp.Add(tx15, fs))
-	require.NoError(t, mp.Add(tx14, fs))
-	err := mp.Add(tx15, fs)
-	require.ErrorIs(t, err, ErrConflictsAttribute)
+	err := mp.Add(tx14, fs)
+	require.Error(t, err)
+	require.True(t, strings.Contains(err.Error(), fmt.Sprintf("conflicting transactions have bigger or equal network fee: %d vs %d", smallNetFee, smallNetFee)))
```
Comment (roman-khimov) [inline pkg/core/mempool/mem_pool_test.go:554]: Welcome to the `errors.Join()` lovers club.

**Review by roman-khimov — APPROVED** [review]

File: pkg/core/mempool/mem_pool_test.go (hunk 2 above)
Comment (AnnaShaleva) [reply pkg/core/mempool/mem_pool_test.go:555]: Looking for bumping minimum required version up to go 1.20.

File: pkg/core/mempool/mem_pool_test.go (hunk 1 above)
Comment (AnnaShaleva) [reply pkg/core/mempool/mem_pool_test.go]: Fixed.

**Pull request merged**